Modify plot noisemodel and plot_toothpick_detail

galaxyumi opened the pull request:
I made some modification to these two plotting codes to (hopefully) improve figures.

For 'plot_noisemodel', I replaced the  'color' parameter with 'cmap_name' and removed the 'label' parameter. This enables to automatically cycle colors for source density bins for a given color map (default='jet'), as well as to automatically plot a legend for any given number of source density bins. 

For 'plot_toothpick_detail', I made the figure size flexiable based on a given number of filters.

karllark commented on beast/plotting/plot_noisemodel.py:
```diff
@@ -13,8 +13,7 @@ def plot_noisemodel(
     samp=100,
-    color=["black", "red", "gold", "lime", "xkcd:azure"],
-    label=None,
+    cmap_name='jet',
```
Maybe better to pick a different cmap as the default?  See https://matplotlib.org/stable/tutorials/colors/colormaps.html

galaxyumi replied:
changed it to 'viridis'

karllark reviewed (changes requested):
Looks good.  But I am not the expert on plot_noisemodel.   Will ask another for a review of this PR too.

christinawlindberg commented:
Almost everything looks good to me. The one caveat is that the labels correspond with the order the noise model files are read in, which isn't very informative. E.g. if I gather up the noise model files with glob, the ordering is 0, 1, 10, 11, 12, 2, 3, 4, 5, 6, 7, 8, 9 which then gets labeled as 0, 1, 2, 3, 4, 5... Also, if there are any missing noise files, the labels shift, and it gets kinda confusing. I typically just label them according to the number in the actual file name, 

`num_label = int(nfile.split(".")[-2].split("bin")[-1])`

P.S. One minor thing I noticed is how hard this plot is to create from a command line if there are multiple noise model files. I don't know how often someone wants to run this from the command line though.

christinawlindberg commented:
@galaxyumi I was only able to make it work with one noise model file, which obviously doesn't convey a lot of information...

galaxyumi commented:
> Almost everything looks good to me. The one caveat is that the labels correspond with the order the noise model files are read in, which isn't very informative. E.g. if I gather up the noise model files with glob, the ordering is 0, 1, 10, 11, 12, 2, 3, 4, 5, 6, 7, 8, 9 which then gets labeled as 0, 1, 2, 3, 4, 5... Also, if there are any missing noise files, the labels shift, and it gets kinda confusing. I typically just label them according to the number in the actual file name,
> 
> `num_label = int(nfile.split(".")[-2].split("bin")[-1])`
> 


Thanks for pointing this out. I added new lines to sort the input noisemodel files and use the correct SD bin numbers for labeling.

karllark commented on beast/plotting/plot_noisemodel.py:
```diff
@@ -190,18 +172,10 @@ def plot_noisemodel(
-        nargs="+",
-        default=None,
-        help="if set, use these labels in a legend for each item in noise_file_list",
+        default="jet",
```
Should be viridis, right?

galaxyumi replied:
Thanks! Fixed it.

karllark reviewed (changes requested):
Look great.  One small change requested.

karllark reviewed (approved):
Looks great.
